### PR TITLE
fix(ext/node): add stubbed process.report

### PR DESCRIPTION
**Pull request by Danielduel**
Targets #21355, #21304 

Issues with vite are caused by the missing node's process report object.
I've stubbed most of it for starter and replaced trivial spots with stubs.
I am not sure how to run compiled binary with built-in node polyfills, but things like `--no-remote` or declaring those deps in the import map and leading them to the deno's source doesn't work.

`cargo build -vv` produces build without a problem
`cargo test` hangs on build step and I need help with that (whole system hang, no output in journalctl)

`tools/format.js` and `tools/lint.js` tell me that `test_util/std/path/mod.ts` can't be found.

### CODE REVIEW

**Comment by bartlomieju** [comment]
> tools/format.js and tools/lint.js tell me that test_util/std/path/mod.ts can't be found.

Run `git submodule update --init --recursive` in the root of the repository and make sure after that `git status` is clean.

**Comment by Danielduel** [comment]
Pushed format fix, `cargo test` after `rustup upgrade` and `cargo clean` started to work and it looks like it fails in the same way like the main does.

File: ext/node/lib.rs [hunk 1]
```diff
@@ -350,6 +350,7 @@ deno_core::extension!(deno_node,
     "_next_tick.ts",
     "_process/exiting.ts",
     "_process/process.ts",
+    "_process/report.ts",
```
Comment (kt3k) [inline ext/node/lib.rs:353]: I think the path should be `internal/process/report.ts` to align the file structure as close to Node.js `lib` directory as possible. https://github.com/nodejs/node/tree/main/lib/internal/process

Comment (Danielduel) [reply ext/node/lib.rs:353]: I wasn't aware of it, will do!

**Comment by Danielduel** [comment]
I would like to ask for another round of reviews @bartlomieju @kt3k

**Comment by bartlomieju** [comment]
@Danielduel could you try your local Deno build with this change and check if it solves the problem from the reported issue?

**Comment by Danielduel** [comment]
> @Danielduel could you try your local Deno build with this change and check if it solves the problem from the reported issue?

Hi @bartlomieju,
I've mentioned it before that I am not sure why it still resolves the wrong module, but now after tinkering with this stuff - I can put into words...

I've done `alias deno=...pathtobuiltdeno`

```ts
import { report as importReport, arch as importArch } from "node:process";
import { createRequire } from "node:module";

console.log("Imported:")
console.log(importReport, importArch);

const require = createRequire(import.meta.url);
const { report, arch } = require("node:process");
console.log("Required:");
console.log(report, arch)
```

Imported deps are fine - report is defined, arch is an empty string.
Report from required is undefined, but arch is "x64" for me.

**Comment by Danielduel** [comment]
I have a feeling that `ext/node/polyfills/01_require.js` Module._load can give me some hints why it is.

**Comment by Danielduel** [comment]
I can confirm that this change will fix vite start.
image

**Review by bartlomieju — COMMENTED** [review]
@Danielduel that's great to see!

Could you add some basic tests to `cli/test/unit_node/process_test.ts` that checks that this function is present and asserting some of the values? Especially the ones that are currently marked as `undefined`.

**Comment by Danielduel** [comment]
Removed reexport since node doesn't seem to provide anything else than a getter on `process.report`.
I will have to fix few typings in DefinitelyTyped regarding `getReport`.
I would like to get feedback if this is how I should write tests for `getReport`.
I haven't ran anything yet, pushing it as a wip to get feedback.

**Review by kt3k — APPROVED** [review]
LGTM!

**Comment by kt3k** [comment]
Could you fix the lint errors like the below?:

```
 5 | function writeReport(filename: string, err: Error) {
   :                                             ^^|^^
   :                                               `-- Instead use the equivalent from the `primordials` object
```

primordials are available like:
```js
const primordials = globalThis.__bootstrap.primordials;
const { Error } = primordials;
```

**Comment by Danielduel** [comment]
I've updated everything that was calling for primodial... with a primodial.
I am not sure if those typings of `typeof Error` are best way to use the Error primodial.
I've figured out how to selectively run my tests with latest binary, I had few mistakes and I've fixed them.
Built deno and retested vite - works.
Linted and formatted the codebase.

**Review by kt3k — APPROVED** [review]
Thanks for updating!

LGTM. Nice work!

**Pull request merged**